packages/server/graphql/mutations/helpers/tiptap/extractTextFromTipTapJSONContent.ts

```diff
@@ -0,0 +1,17 @@
+import {JSONContent} from '@tiptap/core'
+
+const extractTextFromTipTapJSONContent = (input: JSONContent): string => {
+  if (input.text) {
```

Comment:
The methods are defined on class `Editor` (which I found confusing as it's more about the state of the editor, so I renamed it in our code), not on `JSONContent`. We [call](https://github.com/ParabolInc/parabol/pull/6392/files#diff-9a4772bf006047ee4c0e2f50318333ead4d455b37ea159ef99cbac2ce69774a5R80) `getJSON()` on it to get the JSON representation of the content in the editor, and the stringinified version of that object is passed into this mutation. 
I guess we could also call `getText()` there and pass both of the JSON structure and the simple text to this mutation?

Reply:
on the server, rehydrate the JSONContent into an Editor & then call getText on it

Reply:
rehydration sounds good but seems like `Editor` doesn't really work in server env.

When creating it that way
```
  const editor = new Editor({
    element: null,
    content,
    extensions: createEditorExtensions()
  })
```
we get 
```
  "errors": [
    {
      "message": "document is not defined",
      "locations": [
        {
          "line": 33,
          "column": 3
        }
      ],
      "path": [
        "upsertTeamPromptResponse"
      ]
    }
  ],
  "data": null
}
```

It also doesn't work with JSDOM:
```
  const {document} = new JSDOM().window
  const editor = new Editor({
    element: document.createElement('tmp'),
    content,
    extensions: createEditorExtensions()
  })
```

@mattkrick, @jmtaber129 @JimmyLv any ideas?

Reply:
@BartoszJarocki  Directly importing `generateText` seems to work for me:
```
import {JSONContent, generateText} from '@tiptap/core'
import StarterKit from '@tiptap/starter-kit'

const extractTextFromTipTapJSONContent = (input: JSONContent): string => {
  return generateText(input, [StarterKit], {})
}

export default extractTextFromTipTapJSONContent
```

<img width="1106" alt="Screen Shot 2022-04-27 at 12 56 26 PM" src="https://user-images.githubusercontent.com/9013217/165560773-2892c57c-a27d-491b-96fa-65aefd7e8d49.png">

Reply:
oh this is great find! I don't see that in their docs but seems to do exactly what we need: https://github.com/ueberdosis/tiptap/blob/main/packages/core/src/helpers/generateText.ts 

thank you @jmtaber129 🙏

Reply:
@mattkrick done. i also extracted a helper method to create an array of extensions just to make sure we use the same config on client/backend

Reply:
Yea, I was suspicious of using it at first since it's in the `helpers` folder, but it's used in an example added in the PR - https://github.com/ueberdosis/tiptap/pull/1875/files#diff-ce42067e2376ec11487bcf3eb3cfc63fe82c8c1fe236794954e7a5e01f4cc72aR46

Reply:
yeah, should be safe to use. great find!